packages/holocron-dev-server/__tests__/src/utils/module-map.spec.js

```diff
@@ -0,0 +1,214 @@
+});
+
+describe('loadRemoteModuleMap', () => {
+  test('returns json response from loading module map', async () => {
+    fetchMockFn.mockResolvedValueOnce({
+      ok: true,
+      json: jest.fn(() => Promise.resolve({
+        modules: {
+          'child-module': {
+            browser: {
+              url: 'https://example.com/modules/child-module/child-module.js',
+            },
+          },
+        },
+      })
+      ),
+    });
+    const remoteModuleMapUrl = 'https://example.com/modules/module-map.json';
+    const remoteModuleMap = await loadRemoteModuleMap(remoteModuleMapUrl);
+    expect(remoteModuleMap).toEqual({
+      modules: {
+        'child-module': {
+          browser: {
+            url: 'https://example.com/modules/child-module/child-module.js',
+          },
+        },
+      },
+    });
+  });
+  test('remote module map is not string returns empty list of modules ', async () => {
+    const remoteModuleMapUrl = {};
+    const remoteModuleMap = await loadRemoteModuleMap(remoteModuleMapUrl);
+    expect(remoteModuleMap).toEqual({
+      modules: {},
+    });
+  });
+  test('error response if fetching remote module map returns status not ok', async () => {
+    fetchMockFn.mockResolvedValueOnce({
+      ok: false,
+    });
+    const remoteModuleMapUrl = 'https://example.com/modules/module-map.json';
+    const remoteModuleMap = await loadRemoteModuleMap(remoteModuleMapUrl);
+    expect(remoteModuleMap).toEqual({
+      modules: {},
+    });
+  });
+  test('error response if fetching remote module map throws errors', async () => {
+    fetchMockFn.mockRejectedValue(new Error('Async error'));
+    const remoteModuleMapUrl = 'https://example.com/modules/module-map.json';
+    const remoteModuleMap = await loadRemoteModuleMap(remoteModuleMapUrl);
+    expect(remoteModuleMap).toEqual({
+      modules: {},
+    });
+  });
+});
+
+describe('createLocalModuleMap', () => {
+  test('generates an empty local module map without modules', () => {
+    const localModuleMap = createLocalModuleMap();
+    expect(localModuleMap).toEqual({
+      modules: {},
+    });
+  });
+
+  test('generates a local module map from modules', () => {
+    const modules = [
+      {
+        moduleName: 'root-module',
+      },
+    ];
+    const localModuleMap = createLocalModuleMap(modules);
+    expect(localModuleMap).toEqual({
+      modules: {
+        'root-module': {
+          baseUrl: '/static/modules/root-module/',
+          browser: {
+            url: '/static/modules/root-module/root-module.js',
+          },
+        },
+      },
+    });
+  });
+});
+
+describe('createUnifiedModuleMap', () => {
+  test('combines the remote and local module map', () => {
+    const modules = {};
+    const localModuleMap = { modules };
+    const remoteModuleMap = { modules };
+    const moduleMap = createUnifiedModuleMap({
+      localModuleMap,
+      remoteModuleMap,
+    });
+    expect(moduleMap).toEqual({
+      modules: {},
```

Comment:
could you include some modules to show that the two maps are combined

Reply:
nice catch! yes, i've updated this to reflect that